File: myProblems/myProblems_2844/Solution.java
package myProblems.myProblems_2844;

import com.alibaba.fastjson.JSON;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public int minimumOperations(String num) {
        int ans = num.contains("0") ? num.length() - 1 : num.length();
        var ends = new String[]{"00", "25", "50", "75"};
        for (var end : ends) {
            ans = Math.min(ans, help(num, end));
        }
        return ans;
    }
    int help(String num,String end){
        var i = end.charAt(1);
        var j = end.charAt(0);
        int idx_i = num.lastIndexOf(i);
        if (idx_i <= 0)return num.length();
        int idx_j = num.lastIndexOf(j, idx_i - 1);
        if (idx_j < 0)return num.length();
        else return num.length() - idx_j - 2;
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        String num = jsonStringToString(inputJsonValues[0]);
        return JSON.toJSON(minimumOperations(num));
    }
}

File: myProblems/myProblems_2844/solution.rs
use serde_json::{json, Value};

pub struct Solution;

impl Solution {
	pub fn minimum_operations(num: String) -> i32 {
		let mut ans = num.len() as i32;
		if num.contains('0'){ ans -= 1;}
		let ends = vec!["25","50","75","00"];
		ends.iter().for_each(|end|{
			ans = std::cmp::min(ans,Self::help(num.clone(),end.to_string().chars().into_iter().collect::<Vec<char>>()));
		});
		ans
	}
	fn help(num:String,end:Vec<char>) -> i32{
		let n = num.len() as i32;
		if let Some(i) = num.rfind(end[1]){
			return if let Some(j) = num[..i].rfind(end[0]) {
				n - j as i32 - 2
			} else {
				n
			}
		}
		n
	}
}

#[cfg(feature = "solution_2844")]
pub fn solve(input_string: String) -> Value {
	let input_values: Vec<String> = input_string.split('\n').map(|x| x.to_string()).collect();
	let num: String = serde_json::from_str(&input_values[0]).expect("Failed to parse input");
	json!(Solution::minimum_operations(num))
}
